### PR TITLE
Adding OCI Data lake support in OCIFS

### PR DESCRIPTION
**Pull request by ramackri**
To enable customers to load/store data from OCI Data Lake for training and inference, OCI Data Lake needs to be supported like Object Storage via OCIFS package.We should leverage OCIFS to implement the client for their customers to be able to access the lake-managed mount's bucket via OCIFS.
https://confluence.oci.oraclecorp.com/pages/viewpage.action?spaceKey=LAKEHOUSE&title=Access+to+Lake+Data+using+OCIFS
https://confluence.oci.oraclecorp.com/display/LAKEHOUSE/Access+to+Lake+Data+using+ADS+with+OCIFS
Fixes:
1.Adding Lake support for accessing data lake via ocifs
2.To quickly see all messages, you can set the environment variable OCIFS_LOGGING_LEVEL=DEBUG
3.Added minor fix in the core.py for full path
4.Readme file updated for Data lake support
Pending Items(This will be covered as part of new PR):
1.Adding Test cases for Data lake support

### CODE REVIEW

**Comment by bharatviswa504** [comment]
@ramackri can you specify the generated files from SDK which can help reviewers focus on core changes

**Review by ahosler — APPROVED** [review]
This looks great! Thank you for your hard work! Excited to support Data Lake customers

File: ocifs/core.py [hunk 1]
```diff
@@ -325,12 +376,15 @@ def connect(self, refresh=True):
         )
         self._get_region()
         try:
-            self.oci_client = ObjectStorageClient(self.config, **self.config_kwargs)
+            self.oci_client = LakeSharingObjectStorageClient(self.config, **self.config_kwargs)
```
[the remark this answers is not in the document]
Comment (ramackri) [reply ocifs/core.py]: LakeSharingObjectStorageClient will fallback to ObjectStorageClient internally without any issues

File: ocifs/core.py [hunk 2]
```diff
@@ -284,14 +303,46 @@ def split_path(self, path, **kwargs):
...
         full_bucket, _, obj_path = path_sans_protocol.partition("/")
-        bucket, _, namespace = full_bucket.partition("@")
+        # Added the below check for lake support
+        if "@ocid1.lake" in full_bucket:
+            mount_name, _, lake_ocid = full_bucket.partition("@")
+            mount_spc_information = mount_name.split(":")
+            mount_type = get_mount_type(mount_spc_information)
+            logger.debug(f"mount_type:: {mount_type}")
+            if not mount_type:
+                raise ValueError('mount path mentioned in ocifs URI is invalid.Please check the URI format!!!')
```
[the remark this answers is not in the document]
Comment (ramackri) [reply ocifs/core.py]: This error message will be covered as part of Upcoming PR

**Pull request merged**